sycl/test/esimd/vadd.cpp

```diff
@@ -1,5 +1,4 @@
-// TODO ESIMD enable host device under -fsycl
-// RUN: %clangxx -I %sycl_include %s -o %t.out -lsycl
+// RUN: %clangxx -fsycl %s -o %t.out
```

Comment:
Sorry for not catching it, but there is a regression for this test. BTW, I'm not sure how it slipped through the CI.
This test fails with:
```
InvalidFunctionCall: Unexpected llvm intrinsic:
 llvm.genx.local.id.v3i32 [Src: /iusers/dbakhval/github/llvm/llvm-spirv/lib/SPIRV/SPIRVWriter.cpp:2782 false ]
```
As of today, we still require `-fsycl-explicit-simd` flag for ESIMD code. I submitted #3457 to fix this.

Reply:
> BTW, I'm not sure how it slipped through the CI.

This casts shadow on CI validation and ESIMD feature validation in particular. @DenisBakhvalov, could you investigate why this regression happened, please? Could it be a conflict between commits?